specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,306 @@
+
+**nimble-angular**  
+Each i18n provider will have its own Angular directive and module (prototype: [nimble-i18n-core.directive](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core.directive.ts) and [nimble-i18n-core.module](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core.module.ts) for `i18n-core`.)
+
+In order to make it easy/automatic for clients to pick up new localized strings/labels when they uptake new nimble-angular versions, each i18n provider has an additional directive that will set all of the Nimble-defined labels/strings, using Angular's `$localize` function on the English strings.  
+Prototype: [nimble-i18n-core-with-defaults.directive](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core-with-defaults.directive.ts)  
+If we define descriptions for each string, we can include it so it appears in the message files, such as: ``$localize`:Nimble number-field increment button label:Increment` ``.
+
+Once an Angular app uptakes the nimble-angular version that introduces these i18n modules, running `ng extract-i18n` will result in the app pulling in Nimble-provided labels/strings for localization. (Prototype: [messages.xlf](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/locales/messages.xlf), output of `ng extract-i18n` after upgrading to this nimble-angular version)  
+When they pull in new nimble-angular versions in the future and re-run that command, the new strings will again be pulled in for translation automatically.
+
+For each i18n provider that an Angular app will use:
+- The app imports that specific i18n module (prototype: [in example app module](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/app/app.module.ts#L73)).
+- The app adds that i18n element as a child to their theme provider ([prototype](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/app/app.component.html#L2)):
+```html
+<nimble-theme-provider>
+  <nimble-i18n-core withDefaults></nimble-i18n-core>
+</nimble-theme-provider>
+```
+- If the app needs to customize any of the labels, they can do so via the i18n directive API. Generally the root i18n provider would use `withDefaults` to set all the labels to their localized values, and any nested ones would not.
+
+We expect most apps in SystemLink to consume both `nimble-i18n-core` and `nimble-i18n-table` as-is.
```

Comment:
Where do you think this would be included in a SystemLink application? In many cases, the usage of a nimble component comes from a shared library (e.g. sl-lib wraps the nimble-table in the sl-table), and even that component might be wrapped in some other component (e.g. file-grid in the file-manager-lib might wrap the sl-table). However, the `nimble-theme-provider` wraps the entire application content in `app.component.html`. Would an application need to have the knowledge of what shared libraries are doing, or would there be some way to have a component in a library specify these strings?

Reply:
My expectation was that each app would include `nimble-i18n-core` and `nimble-i18n-table` as direct children of `nimble-theme-provider` in their `app.component.html` files.
I don't really want each app to have to be aware of the specific Nimble components being used by the shared library - but since the granularity level is just 'core' and 'table', I think it's reasonable to say 'include core, if you use any [Nimble] grid include table', and apps wouldn't need to revisit that really (i.e. no updates to make if the shared lib depends on more Nimble components later on, since their labels would just be in 'core', generally).

It's theoretically possible to say that `file-grid` could be what includes `nimble-i18n-table`, but then we're back to multiple duplicated i18n providers/ label attributes in the DOM, which is what we wanted to avoid by going with this approach.

If we wanted to enforce that all SLE apps used both i18n providers, then potentially putting them in something like the `sl-swif-header` might be an option too (i.e. some component in the shared lib that's fairly high level and that there's only one of).

Reply:
Updated in the HLD to be more specific and say that each app HTML will include the providers as children of their theme-provider.